phpunit/experimental/modules/class-gutenberg-modules-test.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * Tests Gutenberg_Modules_Test
+ *
+ * @package WordPress
+ */
+
+/**
+ * Tests for the Gutenberg_Modules_Test class.
+ */
+class Gutenberg_Modules_Test extends WP_UnitTestCase {
+
+	public function get_test_data() {
+		return array(
+			'parent_module'         => 'module_1',
+			'parent_module_src'     => 'module_1_src',
+			'dependant_module'      => 'module_2',
+			'dependant_module_src'  => 'module_2_src',
+			'dependant_module_deps' => array(
+				'module_1',
+			),
+			'version'               => '1.0.0',
+			'empty_deps'            => array(),
+		);
+	}
+
+	public function test_register() {
+		$test_data         = $this->get_test_data();
+		$module_identifier = $test_data['parent_module'];
+		$src               = $test_data['parent_module_src'];
+		$dependencies      = $test_data['empty_deps'];
+		$version           = $test_data['version'];
+
+		gutenberg_register_module( $module_identifier, $src, $dependencies, $version );
+
+		// Use reflection to access the private static property $registered.
+		$reflection = new ReflectionClass( 'Gutenberg_Modules' );
```

Comment:
The unit tests included focus on testing internal implementation which will make it harder to refactor code later. How much code would you be able to test without using reflections but through public API?

Reply:
We could move to e2e and test 
 `get_import_map` which uses get_dependencies and enqueue
`print_enqueued_modules`, which uses `get_version_query_string` and needs `register`

With only those two tests we would be covering the entire class.

Reply:
For reference, this is how regular scripts are covered with unit tests in WordPress scripts:

https://github.com/WordPress/wordpress-develop/blob/96339bda237182d2d1246b8b4f12df9833643cd9/tests/phpunit/tests/dependencies/scripts.php

Reply:
I'm afraid we don't have the `assertEqualMarkup` and `parse_markup_fragment` functions in Gutenberg. 

I had to use `WP_HTML_Tag_Processor` instead, just making sure the `src` and the `type` of the scripts enqueued are correct.

I can migrate it to those structure on the backport once is tested in Gutenberg.